snuba/admin/views.py

```diff
@@ -278,6 +279,93 @@ def clickhouse_trace_query() -> Response:
         )
 
 
+@application.route("/clickhouse_querylog_query", methods=["POST"])
+def clickhouse_querylog_query() -> Response:
+    user = request.headers.get(USER_HEADER_KEY)
+    if not user:
+        return Response(
+            json.dumps({"error": "Unauthorized"}),
+            401,
+            {"Content-Type": "application/json"},
+        )
+    req = json.loads(request.data)
```

Comment:
Can you file a tech debt ticket to make this automatic? this is currently very easy to misuse/forget

Reply:
I think it may not be necessary, there is an auth function that runs before every request so I don't think I even needed to do this.

I just added an `if not user` check so mypy won't complain about it possible being `None`, but I could've probably just `assert user is not None` since it won't ever be None in prod.

https://github.com/getsentry/snuba/blob/85c38174832ecad29103b1daac697cd1a78d119f/snuba/admin/views.py#L64-L70